### PR TITLE
upgrade to Terraform 0.14

### PR DESCRIPTION
**Pull request by gidsg**
https://trello.com/c/zZEis2fo/860-3-upgrade-to-terraform-014

Note this version introduces a lock file which fixes plugin versions, it is recommended to check this into version control see: https://www.terraform.io/upgrade-guides/0-14.html#opting-out-of-dependency-locking

This also bumps the AWS provider version which seemed to be a useful way to fix a terraform upgrade error about legacy providers (although possibly just a side-effect of forcing a download of a new provider)

### CODE REVIEW

File: terraform/accounts/main/main.tf [hunk 1]
```diff
@@ -6,7 +6,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = "~> 3.37.0"
```
Comment (bjgill) [inline terraform/accounts/main/main.tf]: Why do we need to bump this version?

Comment (gidsg) [reply terraform/accounts/main/main.tf]: @bjgill I got an error about legacy providers and the upgrade not being able to proceed before I did this. I think maybe the version bump makes it fetch the new provider version which solved the issue as a side-effect.

Comment (bjgill) [reply terraform/accounts/main/main.tf]: Sounds reasonable. Would have been nice to have this information in the commit/PR description.

**Review by bjgill — APPROVED** [review]
:rocket:

**Comment by phantas** [comment]
I was still reviewing but up till now I only had minor suggestions around commit messages so I'll just finish combing the files, that I was in the middle of, but, as far as I can see 👍 from me.

File: terraform/accounts/main/main.tf (hunk 1 above)
Comment (gidsg) [reply terraform/accounts/main/main.tf]: @bjgill fair point, I've updated the PR description now.

**Pull request merged**